### PR TITLE
Add namespace, label, and annotation support to asoctl

### LINKED ISSUE #3865: Support `-n` namespace specifier for asoctl
**Describe the current behavior**
asoctl always imports resources into the default namespace.

**Describe the improvement**
Support a `-n` (or similar) argument to automatically put the imported resources into a specific namespace when applied.

We may also want other simple additions such as a `-l` for "add these labels always" or a `-a` for annotations. Users can do these things with Kustomize too but it might be nice to give them some basic help out of the box.

**Additional context**
This came up as I was writing the asov1->asov2 migration document.

### PR DESCRIPTION
**Pull request by matthchr**
Fixes #3865

**If applicable**:
- [ ] this PR contains documentation
- [x] this PR contains tests
- [ ] this PR contains YAML Samples

### CODE REVIEW

File: v2/cmd/asoctl/cmd/import_azure_resource.go [hunk 1]
```diff
@@ -47,6 +47,25 @@ func newImportAzureResourceCommand() *cobra.Command {
...
 
+	cmd.Flags().StringVarP(
+		&options.namespace,
+		"namespace",
+		"n",
+		"",
+		"Write the imported resources to the specified namespace")
+	cmd.Flags().StringSliceVarP(
+		&options.labels,
+		"label",
```
Comment (matthchr) [inline v2/cmd/asoctl/cmd/import_azure_resource.go:58]: I'm torn between `label` and `labels` here -- same for annotations.

Thoughts?

Comment (theunrepentantgeek) [reply v2/cmd/asoctl/cmd/import_azure_resource.go:58]: Definitely singular. 

The only plural CLI commands I can think of are ones that only ever deal with many things, and these options do need to support singles.

File: v2/cmd/asoctl/cmd/import_azure_resource.go [hunk 2]
```diff
@@ -47,6 +47,25 @@ func newImportAzureResourceCommand() *cobra.Command {
...
+		"l",
+		nil,
+		"Add the specified labels to the imported resources. Multiple comma-separated labels can be specified (example.com/mylabel=foo,example.com/mylabel2=bar) or the --label (-l) argument can be used multiple times (-l example.com/mylabel=foo -l example.com/mylabel2=bar)")
```
Comment (theunrepentantgeek) [inline v2/cmd/asoctl/cmd/import_azure_resource.go:61]: Can you put a space after the comma, e.g.

```
--label example.com/mylabel=foo, example.com/mylabel2=bar
```

Also, the examples need to be consistent:

```suggestion
		"Add the specified labels to the imported resources. Multiple comma-separated labels can be specified (--label example.com/mylabel=foo,example.com/mylabel2=bar) or the --label (-l) argument can be used multiple times (-l example.com/mylabel=foo -l example.com/mylabel2=bar)")
```

File: v2/internal/util/annotations/parse.go [hunk 3]
```diff
@@ -0,0 +1,78 @@
+/*
+ * Copyright (c) Microsoft Corporation.
+ * Licensed under the MIT license.
+ */
+
+package annotations
```
Comment (theunrepentantgeek) [inline v2/internal/util/annotations/parse.go:6]: Putting these into a new package is perfect, but does it need the intermediate `util` folder? Suggest moving these up one level.

File: v2/internal/util/annotations/parse.go [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+
+	// TODO: Could enforce character restrictions too but not bothering for now
+
+	return Annotation{
+		Key:   key,
+		Value: value,
+	}, nil
+}
+
+func ParseAll(annotations []string) ([]Annotation, error) {
```
Comment (theunrepentantgeek) [inline v2/internal/util/annotations/parse.go:66]: Needs a doc-comment.

File: v2/internal/util/labels/parse.go [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+		return Label{}, err
+	}
+
+	return Label{
+		Key:   annotation.Key,
+		Value: annotation.Value,
+	}, nil
+}
+
+func ParseAll(labels []string) ([]Label, error) {
```
Comment (theunrepentantgeek) [inline v2/internal/util/labels/parse.go:35]: Needs a doc-comment.

**Review by theunrepentantgeek — APPROVED** [review]

File: v2/internal/util/annotations/parse.go (hunk 3 above)
Comment (matthchr) [reply v2/internal/util/annotations/parse.go:6]: We do already have a bunch of packages in the `util` folder, but I've moved these out

File: v2/internal/util/annotations/parse.go (hunk 4 above)
Comment (matthchr) [reply v2/internal/util/annotations/parse.go:66]: Fixed

File: v2/internal/util/labels/parse.go (hunk 5 above)
Comment (matthchr) [reply v2/internal/util/labels/parse.go:35]: Fixed

**Pull request merged**